src/main/java/gift/dto/ProductRequest.java

```diff
@@ -1,14 +1,30 @@
 // src/main/java/gift/dto/ProductRequest.java
 package gift.dto;
 
-
+import gift.validation.ValidationGroups;
 import jakarta.validation.constraints.Min;
 import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.Pattern;
+import jakarta.validation.constraints.Size;
 
 public record ProductRequest(
-        @NotBlank(message = "상품 이름은 필수 입력값입니다.") String name,
-        @Min(value = 1, message = "가격은 1원 이상이어야 합니다.") int price,
-        @NotBlank(message = "이미지 URL은 필수 입력값입니다.") String imageUrl
-) {
+        @NotBlank(message = "상품 이름은 필수 입력값입니다.",
+                groups = {ValidationGroups.DefaultGroup.class, ValidationGroups.PermittedGroup.class})
+        @Size(max = 15, message = "상품 이름은 공백 포함 최대 15자까지 입력할 수 있습니다.",
+                groups = {ValidationGroups.DefaultGroup.class, ValidationGroups.PermittedGroup.class})
+        @Pattern(regexp = "^[\\p{IsHangul}\\p{IsHan}\\p{IsAlphabetic}\\d\\s()\\[\\]+\\-&/_]+$",
+                message = "상품 이름에 허용되지 않는 문자가 포함되었습니다.",
+                groups = {ValidationGroups.DefaultGroup.class, ValidationGroups.PermittedGroup.class})
+        @Pattern(regexp = "^(?!.*카카오).*",
```

Comment:
```suggestion
        @Pattern(regexp = ProductNamePatterns.ALLOWED_CHARACTERS,
                message = "상품 이름에 허용되지 않는 문자가 포함되었습니다.",
                groups = {ValidationGroups.DefaultGroup.class, ValidationGroups.PermittedGroup.class})
        @Pattern(regexp =ProductNamePatterns.KAKAO_RESTRICTION,
```
- 정규식 패턴을 다른 곳에서도 활용하고 있던데요. 정책이 분산되어 있으면 나중에 실수 할 수 있어요. 그리고 이런 부분은 static final로 선언해서 재사용하는건 어떨까요?